app/services/payment_requests/create_service.rb

```diff
@@ -10,12 +10,16 @@ def initialize(organization:, params:)
     end
 
     def call
-      # TODO: Return error unless premium license
+      unless License.premium? && organization.premium_integrations.include?("dunning")
+        return result.not_allowed_failure!(code: "premium_addon_feature_missing")
+      end
 
       return result.not_found_failure!(resource: "customer") unless customer
       return result.not_found_failure!(resource: "invoice") if invoices.empty?
 
-      # TODO: Return error if invoices are not overdue
+      if invoices.where(payment_overdue: false).exists?
```

Comment:
```suggestion
      if invoices.not.payment_overdue.exists?
```
Would you use the existing scope?

Or, if you prefer not to, exists? accepts conditions like

```
      if invoices.exists?(payment_overdue: false)
```